lua/auto-session-library.lua

```diff
@@ -100,15 +100,34 @@ end
 
 function Lib.escaped_session_name_from_cwd()
   local cwd = vim.fn.getcwd()
+  if vim.fn.has('win32') then
+    cwd = cwd:gsub(':', '++')
+    if not vim.o.shellslash then
+      cwd = cwd:gsub("\\", "\\%%")
+    end
+
+    return cwd
+  end
+
   return cwd:gsub("/", "\\%%")
 end
 
 function Lib.legacy_session_name_from_cwd()
   local cwd = vim.fn.getcwd()
+  if vim.fn.has('win32') then
+    cwd = cwd:gsub(':', '++')
+    if not vim.o.shellslash then
+      cwd = cwd:gsub("\\", "-")
+    end
+
+    return cwd
+   end
+
   return cwd:gsub("/", "-")
 end
 
 function Lib.is_readable(file_path)
+  if vim.fn.has('win32') then return true end -- Laziest way to make restore session work Windows
```

Comment:
Should you assume the path is actually readable though? Is there not a way to actually check like we do with `vim.fn.filereadable`?

Also you can just reuse the constant `WIN32` from the previous comments